Convert all docs/*.html to .md

levitte opened the pull request:
We only take the easy parts, and leave manual page generation files alone.

Partial fix of #324

levitte committed:
Convert docs/*.html to .md

We only take the easy parts, and leave manual page generation files alone.

Partial fix of #324

levitte commented:
Actually, I had a thought with this one that will need re-approval later

paulidale commented on docs/fips.md:
```diff
@@ -0,0 +1,55 @@
+    created. This Module was designed for compatibility with OpenSSL so
+    that products using the OpenSSL API can be converted to use
+    validated cryptography with minimal effort.
+-   The OpenSSL FIPS Object Module 2.0 validation is \"delivered\" in
```
Spurious \\ here and later

levitte replied:
Fixed most of them

levitte commented:
There were a few more files to fix...  Re-approval needed

levitte commented:
I noticed this work wasn't quite complete, so back to draft

levitte committed:
Convert docs/sub-*.html.tt to .md.tt

This removes docs/mansidebar.html.tt, which is essentially replaced
by docs/sub-dirdata.yaml.tt

levitte commented:
Rebased ~~and squashed~~

levitte committed:
bin/from-tt: Correct bugs for determining [% top %]

levitte committed:
Bugfix: make knows $(dir ...), not $(dirname ...)

levitte commented:
Squashed and generally cleaned up

levitte commented:
https://beta.openssl.org/ is the test bed for this

t8m commented:
I see some _interesting_ files in the directory list at: https://beta.openssl.org/docs/fips.html
- the .md files and dirdata.yaml in particular. Can we somehow filter them out?

levitte committed:
fixup! Convert docs/*.html to .md

levitte commented:
> I see some _interesting_ files in the directory list at: https://beta.openssl.org/docs/fips.html
> 
>     * the .md files and dirdata.yaml in particular. Can we somehow filter them out?

Yes, see ~~568c8e4fa30deb06b83cacb2924ec60f19515ff1~~27d41d0f2891943c27692007f8e70face3e5c3c7

t8m reviewed (commented):
Now the only glitch I see in docs on beta.openssl.org is that there are manual pages only for the master branch. The rest is giving a weird 404 page. For example: https://beta.openssl.org/docs/man3.0

levitte commented:
Ping

t8m commented:
@levitte Is my comment above relevant?

levitte commented:
> Now the only glitch I see in docs on beta.openssl.org is that there are manual pages only for the master branch. The rest is giving a weird 404 page. For example: https://beta.openssl.org/docs/man3.0

The default `make` command only builds the master docs...  I'm running `make docs` now, which builds the whole set...  so er, in 15-20 minutes?

levitte commented:
> > Now the only glitch I see in docs on beta.openssl.org is that there are manual pages only for the master branch. The rest is giving a weird 404 page. For example: https://beta.openssl.org/docs/man3.0
> 
> The default `make` command only builds the master docs... I'm running `make docs` now, which builds the whole set... so er, in 15-20 minutes?

Done

t8m commented:
LGTM

levitte commented:
> LGTM

@openssl/omc, final review?

levitte commented:
Merged

9d6ac3f1db833a495df684429215e31c76c8ea4e Convert docs/*.html to .md
964dd4386ddb005b1dc52eb197ac53f4686041b2 Convert docs/sub-*.html.tt to .md.tt
18aaf09b6e2dc3b4e1194d561059cec9a2339155 bin/from-tt: Correct bugs for determining [% top %]
e0824ffd90afab8425e01c90594c03b13d9d9975 Bugfix: make knows $(dir ...), not $(dirname ...)
